warehouse/query-core/src/main/java/datawave/query/rules/FieldExistenceRule.java

```diff
@@ -95,6 +101,18 @@ public QueryRuleResult validate(QueryValidationConfiguration ruleConfiguration)
         return result;
     }
 
+    private String formatMessage(FieldMissingFromDateRangeVisitor.ImmaterialNode irrelevantJunction) {
+        // @formatter:off
+        return new StringBuilder()
+                        .append("Fields ")
+                        .append(irrelevantJunction.getFields())
+                        .append(" found in the query subset '")
+                        .append(JexlStringBuildingVisitor.buildQuery(irrelevantJunction.getNode()))
+                        .append("' were not ingested in the provided date range.")
+                        .toString();
+        // @formatter:on
+    }
+
```

Comment:
Would it be better if we just output the individual fields that are missing? Thinking it might be a bunch of output if a specific field is used multiple times and might make finding all of the different fields a bit difficult.

Reply:
Alternatively, it could be helpful for the user to see which snippet of the query is made immaterial by the missing fields. Perhaps this is a decision for the customer to weigh in on.